controllers/common/util.go

```diff
@@ -222,21 +222,25 @@ func GetCmOfMapCs(r client.Reader) (*corev1.ConfigMap, error) {
 func GetStorageClass(r client.Reader) (*storagev1.StorageClassList, error) {
 	csStorageClass := &storagev1.StorageClassList{}
 	err := r.List(context.TODO(), csStorageClass)
-	if err != nil {
-		return nil, err
+	if err == nil {
+		return csStorageClass, nil
 	}
-	return csStorageClass, nil
+
+	if errors.IsNotFound(err) {
+		return nil, fmt.Errorf("storageClassList is not found")
+	}
+
```

Comment:
Sorry for the misleading here, I found that `errors.IsNotFound(err)` is not accurate here.

Usually, we will check `errors.IsNotFound(err)` when we want to use `r.Get()` to get some specific resource. It will have the case that the specific resource is not found. But this does not apply to `r.List()` here. Therefore, we could remove it.


And a little advice here.

I am wondering whether we could combine the `ValidateStorageClass` and `GetStorageClass` together called `CheckStorageClass`.

In this case, we don't need to return the `StorageClass`,
and after we List the `storageClass`, check its length directly.
```
err := r.List(context.TODO(), csStorageClass)
if err != nil {
    // hitting error when we try to List storageclass
    // and we indicate what exactly error is
    return fmt.Errorf("fail to list storageClass: %v", err)
}

// after list the storageClass, we check the length and print the proper info
return nil
```

Reply:
Updated, please review, thanks!